holoviews/plotting/plot.py

```diff
@@ -63,6 +63,12 @@ def state(self):
         raise NotImplementedError
 
 
+    def cleanup(self):
+        """
+        Allows defining cleanup actions to perform on plot deletion.
+        """
```

Comment:
Why not raise ``NotImplementedError`` here?

Reply:
Sure.

Reply:
Actually not totally convinced, a ``Plot`` doesn't necessarily have any cleanup to do, so they shouldn't have to define it.

Reply:
Ok. In that case at least put in ``pass`` to make it explicit that something isn't missing.

Reply:
Nevermind the baseclass should actually clean up streams itself so this is a non-issue.